services/QuillLMS/app/services/generate_fullname.rb

```diff
@@ -1,28 +1,23 @@
 # frozen_string_literal: true
 
-class GenerateFullname
-
+class GenerateFullname < ApplicationService
+  DEFAULT_NAME = 'Firstname Lastname'
   def initialize(name)
     @name = name
   end
 
-  def call
+  def run
     generate_fullname
   end
 
   attr_reader :name
   private :name
 
   private def generate_fullname
-    first_name, last_name = SplitName.new(name).call
-
-    if last_name.nil? && first_name.present?
-      return "#{first_name} #{first_name}"
-    end
+    first_name, last_name = SplitName.run(name)
```

Comment:
This is a bit code golf-y, so take or leave, but consider:
```ruby
return name unless last_name.nil?

first_name.present? ? "#{first_name} #{first_name}" : DEFAULT_NAME

Reply:
Thanks.  I think that's more readable.